Sources/BraintreeDataCollector/BTDataCollector.m

```diff
@@ -130,11 +132,11 @@ - (void)collectDeviceData:(void (^)(NSString * _Nonnull))completion {
 
 #pragma mark - Helper methods
 
-+ (NSString *)generatePayPalClientMetadataID {
++ (NSString *)generatePayPalClientMetadataID:(BOOL)isSandbox {
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wundeclared-selector"
-    if (PayPalDataCollectorClass && [PayPalDataCollectorClass respondsToSelector:@selector(generateClientMetadataID)]) {
-        return [PayPalDataCollectorClass performSelector:@selector(generateClientMetadataID)];
+    if (PayPalDataCollectorClass && isSandbox ? [PayPalDataCollectorClass respondsToSelector:@selector(sandboxGenerateClientMetadataID)] : [PayPalDataCollectorClass respondsToSelector:@selector(generateClientMetadataID)]) {
+        return isSandbox ? [PayPalDataCollectorClass performSelector:@selector(sandboxGenerateClientMetadataID)] : [PayPalDataCollectorClass performSelector:@selector(generateClientMetadataID)];
```

Comment:
Because we are using `@selector` methods to call into the PPDataCollectorClass under the hood, we need to have 2 different functions that take no parameters (since selectors cannot send params). 1 func that takes no params and sets the environment as sandbox and 1 that takes no params and sets the environment as live.

Reply:
nit: could we go with a nested if here to reduce the line length? `if (isSandbox)` instead of the ternary operator would tighten it up a bit.